Testscripts/Linux/customKernelInstall.sh

```diff
@@ -302,22 +302,41 @@ function InstallKernel() {
             apt install -y wget
             apt remove -y linux-cloud-tools-common
             wget "$CustomKernel"
-            LogMsg "Installing ${CustomKernel##*/}"
-            dpkg -i "${CustomKernel##*/}"  >> $LOG_FILE 2>&1
-            kernelInstallStatus=$?
-            image_file=$(ls -1 *.deb* | grep -v "dbg" | sed -n 1p)
+            if [[ $CustomKernel = *.gz || $CustomKernel = *.tar ]]; then
+                tar -xf "${CustomKernel##*/}"
```

Comment:
let s do it a little bit different, as it is a generic workflow.

If the kernel files are in a tar.gz or any kind of archive, just extract them before and then follow the same behaviour as before, if the archive contains debs, go for deb install and if it has rpms go for rpm installs. no need to duplicate code and so forth or to check if the url or path itself has deb in it

Reply:
As @ader1990 said, if tar.gz and rpm has the same contents, deb files, please re-use the code.